matlab_simulation/code v1.0/ME597-7-Trajectories/robottraj.m

```diff
@@ -1,85 +1,133 @@
 % Robot trajectories
+clear;
+close;
+clc;
 
 % Time
 Tmax = 10;
 dt = 0.1;
 T = 0:dt:Tmax;
 
-% Spiral
-x0 = [1 1 1]';
-xddot = zeros(3,length(T));
-xd = zeros(3,length(T)+1);
-xd(:,1)= x0;
-v = exp(-0.2*T);
-w = ones(size(T));
-for t=1:length(T)
-    xddot(:,t) = [v(t)*cos(xd(3,t));
-                  v(t)*sin(xd(3,t));
-                  w(t)];
-    xd(:,t+1) = xd(:,t)+dt*xddot(:,t);
-end
+% Let user choose the input trajectory
+disp('What is the input trajectory?')
+disp('1: Spiral; 2: Nudges; 3: Swerves; 4: Corner;')
+disp('5: User defined motion model')
+prompt = '';
+index = input(prompt);
 
-figure(1);clf; hold on;
-plot(T,xd(:,1:end-1));
+% Let user define an initial condition
+disp('Please give an initial condition, i.e. [1 1 1]')
+prompt = '';
+x0 = input(prompt)';
 
-figure(2);clf;hold on;
-plot(xd(1,:),xd(2,:));
-for t=1:3:length(T)
-    drawcar(xd(1,t),xd(2,t),xd(3,t),.05,2);
-end
-title('Desired Trajectory');
-axis equal;
-
-% Squiggle
-x0 = [1 1 1]';
+% Define the states and initial condition
 xddot = zeros(3,length(T));
 xd = zeros(3,length(T)+1);
 xd(:,1)= x0;
-for t=1:length(T)
-    xddot(:,t) = [2*cos(xd(3,t));
-                  1*sin(xd(3,t));
-                  (xd(1,t))];
-    xd(:,t+1) = xd(:,t)+dt*xddot(:,t);
-end
 
-figure(3);clf; hold on;
-plot(xd(1,:),xd(2,:));
-for t=1:3:length(T)
-    drawcar(xd(1,t),xd(2,t),xd(3,t),.2,3);
-end
-title('Desired Trajectory');
-axis equal;
 
-% Motions
-x0 = [1 1 1]';
-xddot = zeros(3,length(T));
-xd = zeros(3,length(T)+1);
-xd(:,1)= x0;
-v = 2*ones(size(T));
-w = zeros(size(T));
-c = floor(length(w)/8);
+if index == 1
+    % Spiral
+    v = sin(0.2*T);
+    w = ones(size(T));
+    
+    for t=1:length(T)
```

Comment:
The iteration of the motion model is very much the same across other examples, factor this for loop out? Same goes to all other examples?